python/dlisio/lis/load.py

```diff
@@ -25,68 +24,117 @@ def load(path):
+    files.
+    """
+    def __init__(self, path, error_handler):
+        self.error_handler = error_handler
+        self.path = path
 
-    f.close()
+        self.logical_files = []
+        self.reel = HeaderTrailer()
+        self.tape = HeaderTrailer()
 
-    logical_files = []
-    reel = HeaderTrailer()
-    tape = HeaderTrailer()
+        self.check_for_tapemarks()
+        self.file = None
 
-    while True:
-        # Open a new file-handle until we hit EOF
+    def check_for_tapemarks(self):
+        """ Checks whether file is TIFed and adjusts initial offset accordingly
+        """
+        initial_offset = 0
+        f = core.open(self.path, initial_offset)
+
+        def read_as_tapemark(f):
+            try:
+                return core.read_tapemark(f)
+            except:
+                f.close()
+                msg = 'Cannot read {} first bytes of file: {}'
+                raise IOError(msg.format(initial_offset + 12, self.path))
+
+        # Read the first 12 bytes of the file assuming TapeImageFormat.
+        # Verify assumption before moving on.
+        tm = read_as_tapemark(f)
+        self.is_tif = core.valid_tapemark(tm)
+
+        # Some TapeImageFormat files start with tapemark(s) of type 1
+        # (EOF-tapemark).
+        # Opening a lfp tapeimage protocol at a type 1 tapemarks causes an
+        # instant EOF. Hence we have to find the offset of the first type 0
+        # tapemark, which will be the start of our logical file.
+        if self.is_tif:
+            while tm.type != 0:
+                tm = read_as_tapemark(f)
+            initial_offset = f.ptell - 12
+
+        self.offset = initial_offset
+        f.close()
+
+    def open_file(self):
+        """ Open new file.
+
+        Returns True if new file was opened. Otherwise returns False or throws.
+        """
         try:
-            f = core.openlis(path, offset, is_tif)
+            self.file = core.openlis(self.path, self.offset, self.is_tif)
+            return True
         except EOFError:
-            break
+            return False
         except OSError as e:
-            msg =  'dlisio.lis.load: stopped indexing at tell {}\n'
-            msg += 'Reason: {}\nFilepath: {}'
-            logging.error(msg.format(offset, e, path))
-            break
-
-        index = f.index_records()
-
-        # If not a single record could be indexed, close the file and stop here
-        if index.size() == 0 and index.isincomplete():
-            f.close()
-            break
+            self.error_handler.log(
+                core.error_severity.critical,
+                "dlisio.lis.load: file {}".format(self.path),
+                e,
+                "",
+                "Indexing stopped",
+                "Physical tell: {} (dec)".format(self.offset))
+            return False
+
+    def index_file(self):
+        """ Index current file.
+
+        Returns True if indexing was successful. Otherwise returns False or
```

Comment:
Not super fan of this return true false thing. Feel free to ignore, but I propose:
```
while not indexer.isdone():
     try:
         indexer.index_file(): # Open and index
     except:
         indexer.close()
         raise